CodeChat_Server/windows_installer.ps1

```diff
@@ -0,0 +1,205 @@
+# Get-Command -all will search for any python executable (denoting an installation)
+$PythonVersion = Get-Command -all python -erroraction 'silentlycontinue'
+# If the command does not return a string (with the python.exe) we have to do some stuff
+if([string]::IsNullOrEmpty($pythonVersion)){
+    # Asking the user if they would like to have python installed on their system
+    $response = Read-Host -Prompt "You do not have Python installed, would you like to install the required version ($pythonVersionReq) (y/n)?"
+    if($response -eq "y"){
+        echo "Downloading Python Version $PyTarget"
+        # Downloading Python Unsure if ``-UseDefaulCredential`` is necessary
+        # Using Invoke-WebRequest because I couldn't get the other method outlined in the `tutorial <https://lazyadmin.nl/powershell/download-file-powershell>`_ working guide says it's faster though...
+        Invoke-WebRequest -Uri $PythonURL -OutFile $PythonDest -UseDefaultCredential
+        echo "Python Version $PyTarget Downloaded
+        Installing Python"
+        Read-Host "Press any button to continue, then click 'Install Now' in popup"
+        # 
+        # Redirecting to downloads so that we can run our installation
+        cd $env:USERPROFILE\Downloads
+        # 
+        # Installs Python ``Out-Null`` option makes sure the program doesn't continue before the installation completes, The ``/quietly`` just makes no window pop up, which is fine because ``PrependPath=1`` will select the PATH checkbox for us.  
+        # Potentially Install-Package could be a better fit, not sure if it only works on Powershell Modules or other exe's too
+        Start-Process ".\python-$PyTarget-amd64.exe" -ArgumentList /quietly,PrependPath=1 -Wait
+        echo "Python Version $PyTarget Installed Successfully
+        "
+    }
+    # If they don't have python and they don't want us to do it all we can do is let them know what the minimum version to get is
+    else{
+        echo "Please install the minimum required version of python ($pythonVersionReq) or later
+        (Be sure to add to the PATH)"
+        exit
+    }
+}
+# Returning to User Dir
+cd $env:USERPROFILE
+# |
+# 
+# Grabbing the version of python and splitting it up so as to make parsing easier
+$pythonVersion = python --version
+$pythonVersion = $pythonVersion.Split()[1]
+$pythonVersionMaj = $pythonVersion.Split('.')[0]
+$pythonVersionMin = $pythonVersion.Split('.')[1]
+
+# Version Checking
+# if the version we have is not greater than or equal to the required version, an updated version must be installed
+# checking negative case so that if it evaluates as true (version is less than required) then throw warning
+$InstallBool = False;
+if ($pythonVersionMaj -lt $pythonVersionReqArray[0]){
+    $InstallBool = True
+}
+elseif($pythonVersionMaj -eq $pythonVersionReqArray[0]){
+    if($pythonVersionMaj -lt $pythonVersionReqArray[1]){
+        $InstallBool = True
+    }
+}
+
+if($InstallBool){
```

Comment:
Duplicate code -- combine with above.

Reply:
Do you still want the user to be offered the option to download Python automatically via the script? or is this too potentially fraught with bugs, and they should just get an error message and link to the microsoft store?

Reply:
I'd suggest a prompt -- ask the user if they want to install Python automatically, or abort the install. AFAIK, the existing install code worked (but I didn't test it...)

Reply:
Sounds good.